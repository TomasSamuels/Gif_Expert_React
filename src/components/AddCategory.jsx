import { useState } from 'react';

export const AddCategory = ( { onNewCategory } ) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChanged = ({ target }) =>{
        setInputValue( target.value );
    };

    const onSubmit = ( event ) => {
        event.preventDefault();
        const value = inputValue.trim();
        if( value.length <= 1 ) return;
        onNewCategory( value )
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit }>
            <input
                type="text"
                placeholder="Search gifs"
                value={ inputValue }
                onChange={ onInputChanged }
            />
        </form>
    )
}
