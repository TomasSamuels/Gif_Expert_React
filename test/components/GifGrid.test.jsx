import { render, screen } from "@testing-library/react";
import { GifGrid } from "../../src/components/GifGrid";
import { useFetchGifs } from "../../src/hooks/useFetchGifs";

jest.mock('../../src/hooks/useFetchGifs');

describe('Test GifGrid', () => { 

    const category = 'Attack On Titan';
    const gifts = [
        {
            id: 'ABC',
            title: 'Attack on Titan',
            url: 'https://test.com/aot.jpg'
        },
        {
            id: '123',
            title: 'Eren',
            url: 'https://test.com/eren.jpg'
        }
    ];

    test('should match with the snapshot', () => { 

        useFetchGifs.mockReturnValue({
            images: gifts,
            isLoading: false
        });

        const { container } = render( <GifGrid category={ category } /> );
        expect( container ).toMatchSnapshot();
     });

    test('should show loading on init', () => { 

        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true
        });

        render(<GifGrid category={ category }/>);
        expect( screen.getByText('Loading...') );
        expect( screen.getByText( category ) );
     });

     test('should show images when load in useFetchGifs', () => { 

        useFetchGifs.mockReturnValue({
            images: gifts,
            isLoading: false
        });

        render(<GifGrid category={ category }/>);

        expect( screen.getAllByRole('img').length ).toBe(2);

      });

 });