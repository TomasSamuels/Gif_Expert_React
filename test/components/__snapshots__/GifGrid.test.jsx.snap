// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGrid should match with the snapshot 1`] = `
<div>
  <h3>
    Attack On Titan
  </h3>
  <div
    class="card-grid"
  >
    <div
      class="card"
    >
      <img
        alt="Attack on Titan"
        src="https://test.com/aot.jpg"
      />
      <p>
        Attack on Titan
      </p>
    </div>
    <div
      class="card"
    >
      <img
        alt="Eren"
        src="https://test.com/eren.jpg"
      />
      <p>
        Eren
      </p>
    </div>
  </div>
</div>
`;
